ui/desktop/src/components/settings/ConfigureExtensionModal.tsx

```diff
@@ -0,0 +1,175 @@
+import React from 'react';
+import { Modal, ModalContent, ModalHeader, ModalTitle } from '../ui/modal';
+import { Button } from '../ui/button';
+import { FullExtensionConfig, replaceWithShims } from '../../extensions';
+import { getApiUrl, getSecretKey } from '../../config';
+import { showToast } from '../ui/toast';
+
+interface ConfigureExtensionModalProps {
+  isOpen: boolean;
+  onClose: () => void;
+  onSubmit: () => void;
+  extension: FullExtensionConfig | null;
+}
+
+export function ConfigureExtensionModal({
+  isOpen,
+  onClose,
+  onSubmit,
+  extension,
+}: ConfigureExtensionModalProps) {
+  const [envValues, setEnvValues] = React.useState<Record<string, string>>({});
+  const [isSubmitting, setIsSubmitting] = React.useState(false);
+
+  // Reset form when dialog closes or extension changes
+  React.useEffect(() => {
+    if (!isOpen || !extension) {
+      setEnvValues({});
+    }
+  }, [isOpen, extension]);
+
+  const handleExtensionConfigSubmit = async (e: React.FormEvent) => {
+    e.preventDefault();
+
+    if (!extension) return;
+
+    setIsSubmitting(true);
+    try {
+      // First store all environment variables
+      if (extension.env_keys?.length > 0) {
+        for (const envKey of extension.env_keys) {
```

Comment:
we definitely need a rust refactor to support this kind of pattern, every new key creates a new osx approval flow. will figure one out